publish_courses/tests/test_api.py

```diff
@@ -1,11 +1,111 @@
-# Test TODOs
-# BulkPublishListCreate:
-# invalid account raises permission denied
-# invalid LTI session raises permission denied
-# returns data as expected on successful enqueue()
-# exception raises 500
-
-# SummaryList:
-# invalid LTI session raises permission denied
-# returns data as expected on successful enqueue()
-# exception raises 500
+import json
+from mock import patch, Mock
+import unittest
+
+from django.test import TestCase, RequestFactory
+import publish_courses.api as api
+from rest_framework.exceptions import PermissionDenied
+
+
+class ApiTestCase(TestCase):
+
+    def setUp(self):
+        super(ApiTestCase, self).setUp()
+
+        self.post_data = {
+            'account': 'sis_account_id:school:abc',
+            'term': 'sis_term_id:2015-1'
+        }
+        self.lti_data = {
+            'custom_canvas_user_login_id': '1',
+            'custom_canvas_account_sis_id': 'school:abc',
+        }
+        self.get_data = {
+            'account': 'sis_account_id:school:abc',
+            'term': 'sis_term_id:2015-1'
+        }
+        self.user = Mock(is_authenticated=Mock(return_value=True))
+        self.post_url = '/publish_courses/api/jobs'
+        self.get_url = '/publish_courses/api/show_summary'
+
+    def _prep_request_and_post(self, data):
+        request_kwargs = {
+            'content_type': 'application/json',
+            'data': json.dumps(self.post_data),
+        }
+
+        self.request = RequestFactory().post(self.post_url, **request_kwargs)
+        # these values don't matter; the keys need to be present
+        # for the authorization decorator patches to work
+        self.request.LTI = self.lti_data
+        self.request.data = data
+        self.request.user = self.user
+        bulk_publish = api.BulkPublishListCreate()
+        bulk_publish.request = self.request
```

Comment:
initializing the view with arguments `request=self.request` will do the same thing here.